File: src/main/kotlin/demo/AppConfig.kt
package demo

import akka.actor.ActorSystem
import com.typesafe.config.ConfigFactory
import demo.di.SpringExtension
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.SpringBootConfiguration
import org.springframework.context.ApplicationContext
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.PropertySource
import org.springframework.core.annotation.Order

@SpringBootConfiguration
@PropertySource("classpath:springboot.properties")
open class AppConfig {

    @Autowired
    private lateinit var applicationContext: ApplicationContext

    @Bean
    @Order(-100)
    open fun springExtension(): SpringExtension = SpringExtension(applicationContext)

    @Bean
    open fun actorSystem(): ActorSystem = ActorSystem.create("demo", ConfigFactory.load("akka"))
}

File: common/src/main/scala/demo/spring/AppConfig.scala
package demo.spring

import demo.spring.akka.AkkaAutoConfiguration
import org.springframework.boot.SpringBootConfiguration
import org.springframework.context.annotation.{Import, PropertySource}

@SpringBootConfiguration
@Import(Array(classOf[AkkaAutoConfiguration]))
@PropertySource(Array("classpath:springboot.properties"))
class AppConfig
